### DISCUSSION TITLE
Make currentRange for all AxisModel implementations public to receive and private to modify

### DISCUSSION THREAD

**Discussion by asynchaizer** [KoalaPlot/koalaplot-core#93] [Ideas]
Hi @gsteckman!

We can add a currentRange open to receive, because some custom calculations can be based on this value, for example
1. Set the currentRange to the end of the chart if the right border of the current currentRange is not set at the end - this is useful for real-time charts
2. Make an animation from the current currentRange to the target currentRange, for example, we have a list of some zones on the diagram and when we click on the list item, we want to move the user from one section of the diagram to another and do it with animation
3. Make an analysis of the visible area, for example, there is a data sample of 5000 points and the current viewing area is in the range from 100 to 500 and we want to generate an analysis of the values of a section of this area for the user, show the minimum and maximum values, show the average value for the current viewing area, etc.

Accordingly, we can make the currentViewRange field public separately, or simply open this field for reading, but closed for writing:
```
public var currentRange: ClosedFloatingPointRange<Float> by mutableStateOf(range)
    private set
```

What do you think about it?

**Comment by gsteckman** [comment]
I think that makes sense. To match with the setViewRange function the variable name should be changed to viewRange, and to be consistent it should also be a member of the LinearAxisModel interface.

**Comment by gsteckman** [comment]
I pushed a commit yesterday to expose a viewRange property. Please give it a try.

**Reply by andreas-repp** [reply]
Do you already know when this would be released?

**Reply by andreas-repp** [reply]
I just saw the new release! Thanks!

**Comment by dlozanovic** [comment]
Is there a reason why viewRange is State<ClosedRange < Double > > instead of State<ClosedRange < T > >?

**Reply by gsteckman** [reply]
Yes, for the discrete axes (Int and Long), if the viewRange was a whole number then the panning doesn't work because when you drag with the mouse it fires many events each with a movement less than 1. Also, even if you came up with a way to accumulate the fractional drags, then it would look like nothing moves until you cross a pan distance of 1, and it would jump to the next tick.
